src/pydicom/pixels/utils.py

```diff
@@ -223,6 +231,339 @@ def as_pixel_options(ds: "Dataset", **kwargs: Any) -> dict[str, Any]:
+    arr : numpy.ndarray, optional
+        Compress the uncompressed pixel data in `arr` and use it
+        to set the *Pixel Data*. If `arr` is not used then the existing
+        uncompressed *Pixel Data* in the dataset will be compressed instead.
+        The :attr:`~numpy.ndarray.shape`, :class:`~numpy.dtype` and
+        contents of the array should match the dataset.
+    encoding_plugin : str, optional
+        Use `encoding_plugin` to compress the pixel data. See the
+        :doc:`user guide </old/image_data_compression>` for a list of
+        plugins available for each UID and their dependencies. If not
+        specified then all available plugins will be tried (default).
+    encapsulate_ext : bool, optional
+        If ``True`` then force the addition of an extended offset table.
+        If ``False`` (default) then an extended offset table
+        will be added if needed for large amounts of compressed *Pixel
+        Data*, otherwise just the basic offset table will be used.
+    jls_error : int, optional
+        **JPEG-LS Near Lossless only**. The allowed absolute compression error
+        in the pixel values.
+    j2k_cr : list[float], optional
+        **JPEG 2000 only**. A list of the compression ratios to use for each
+        quality layer. There must be at least one quality layer and the
+        minimum allowable compression ratio is ``1``. When using multiple
+        quality layers they should be ordered in decreasing value from left
+        to right. For example, to use 2 quality layers with 20x and 5x
+        compression ratios then `j2k_cr` should be ``[20, 5]``. Cannot be
+        used with `j2k_psnr`.
+    j2k_psnr : list[float], optional
+        **JPEG 2000 only**. A list of the peak signal-to-noise ratios (in dB)
+        to use for each quality layer. There must be at least one quality
+        layer and when using multiple quality layers they should be ordered
+        in increasing value from left to right. For example, to use 2
+        quality layers with PSNR of 80 and 300 then `j2k_psnr` should be
+        ``[80, 300]``. Cannot be used with `j2k_cr`.
+    **kwargs
+        Optional keyword parameters for the encoding plugin may also be
+        present. See the :doc:`encoding plugins options
+        </guides/encoding/encoder_plugin_options>` for more information.
+    """
+    from pydicom.dataset import FileMetaDataset
+    from pydicom.pixels import get_encoder
+
+    # Disallow overriding the dataset's image pixel module element values
+    for option in _IMAGE_PIXEL.values():
+        kwargs.pop(option, None)
+
+    uid = UID(transfer_syntax_uid)
+    encoder = get_encoder(uid)
+    if not encoder.is_available:
+        missing = "\n".join([f"    {s}" for s in encoder.missing_dependencies])
+        raise RuntimeError(
+            f"The pixel data encoder for '{uid.name}' is unavailable because its "
+            f"plugins are all missing dependencies:\n{missing}"
```

Comment:
Remove "all"? Sounds a bit awkward to me, but maybe that's just me...

Reply:
...because all of its plugins are missing dependencies?